File: go-sdk/pkg/models/operations/deletecollaborationwhitelistentriesid.go
package operations

import (
	"openapi/pkg/models/shared"
	"openapi/pkg/utils"
)

type DeleteCollaborationWhitelistEntriesIDPathParams struct {
	CollaborationWhitelistEntryID string `pathParam:"style=simple,explode=false,name=collaboration_whitelist_entry_id"`
}

type DeleteCollaborationWhitelistEntriesIDRequest struct {
	Retries    *utils.RetryConfig
	PathParams DeleteCollaborationWhitelistEntriesIDPathParams
}

type DeleteCollaborationWhitelistEntriesIDResponse struct {
	ClientError *shared.ClientError
	ContentType string
	StatusCode  int64
}

File: ts-sdk/src/sdk/models/operations/deletecollaborationwhitelistentriesid.ts
import { SpeakeasyMetadata, SpeakeasyBase, RetryConfig } from "../../../internal/utils";
import * as shared from "../shared";



export class DeleteCollaborationWhitelistEntriesIdPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=collaboration_whitelist_entry_id" })
  collaborationWhitelistEntryId: string;
}


export class DeleteCollaborationWhitelistEntriesIdRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  retries?: RetryConfig;

  @SpeakeasyMetadata()
  pathParams: DeleteCollaborationWhitelistEntriesIdPathParams;
}


export class DeleteCollaborationWhitelistEntriesIdResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  clientError?: shared.ClientError;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}
